Sources/Subprocess/Configuration.swift

```diff
@@ -640,84 +651,277 @@ internal struct TrackedFileDescriptor: ~Copyable {
+        }
+    }
+
+    internal var _readFileDescriptor: IODescriptor?
+    internal var _writeFileDescriptor: IODescriptor?
 
     internal init(
-        readFileDescriptor: consuming TrackedFileDescriptor?,
-        writeFileDescriptor: consuming TrackedFileDescriptor?
+        readFileDescriptor: consuming IODescriptor?,
+        writeFileDescriptor: consuming IODescriptor?
     ) {
         self._readFileDescriptor = readFileDescriptor
         self._writeFileDescriptor = writeFileDescriptor
     }
 
-    mutating func readFileDescriptor() -> TrackedFileDescriptor? {
+    mutating func readFileDescriptor() -> IODescriptor? {
         return self._readFileDescriptor.take()
     }
 
-    mutating func writeFileDescriptor() -> TrackedFileDescriptor? {
+    mutating func writeFileDescriptor() -> IODescriptor? {
         return self._writeFileDescriptor.take()
     }
 
-    internal init(closeWhenDone: Bool) throws {
-        let pipe = try FileDescriptor.ssp_pipe()
+    internal init(closeWhenDone: Bool, purpose: Purpose) throws {
+        #if canImport(WinSDK)
+        /// On Windows, we need to create a named pipe.
+        /// According to Microsoft documentation:
+        /// > Asynchronous (overlapped) read and write operations are
+        /// > not supported by anonymous pipes.
+        /// See https://learn.microsoft.com/en-us/windows/win32/ipc/anonymous-pipe-operations
+        while true {
+            /// Windows named pipes are system wide. To avoid creating two pipes with the same
+            /// name, create the pipe with `FILE_FLAG_FIRST_PIPE_INSTANCE` such that it will
+            /// return error `ERROR_ACCESS_DENIED` if we try to create another pipe with the same name.
+            let pipeName = "\\\\.\\pipe\\LOCAL\\subprocess-\(purpose)-\(PipeNameCounter.nextValue())"
+            var saAttributes: SECURITY_ATTRIBUTES = SECURITY_ATTRIBUTES()
+            saAttributes.nLength = DWORD(MemoryLayout<SECURITY_ATTRIBUTES>.size)
+            saAttributes.bInheritHandle = true
+            saAttributes.lpSecurityDescriptor = nil
+
+            let parentEnd = pipeName.withCString(
+                encodedAs: UTF16.self
+            ) { pipeNameW in
+                // Use OVERLAPPED for async IO
+                var openMode: DWORD = DWORD(FILE_FLAG_OVERLAPPED | FILE_FLAG_FIRST_PIPE_INSTANCE)
+                switch purpose {
+                case .input:
+                    openMode |= DWORD(PIPE_ACCESS_OUTBOUND)
+                case .output:
+                    openMode |= DWORD(PIPE_ACCESS_INBOUND)
+                }
+
+                return CreateNamedPipeW(
+                    pipeNameW,
+                    openMode,
+                    DWORD(PIPE_TYPE_BYTE | PIPE_READMODE_BYTE | PIPE_WAIT),
+                    1, // Max instance,
+                    DWORD(readBufferSize),
+                    DWORD(readBufferSize),
+                    0,
+                    &saAttributes
+                )
+            }
```

Comment:
OK, one final thing about this code (sorry). I think we should at least add `PIPE_REJECT_REMOTE_CLIENTS` to the flags.

Ideally I think we'd specify an ACL for the named pipe, to make it harder to interfere with this code by having some other process call `CreateFile` between the `CreateNamedPipeW` and `CreateFileW` calls here.